api/src/main/java/org/geysermc/geyser/api/item/custom/CustomItemData.java

```diff
@@ -144,6 +226,24 @@ interface Builder {
 
         Builder tags(@Nullable Set<String> tags);
 
+        Builder stackSize(@NonNegative int stackSize);
+
+        Builder maxDamage(int maxDamage);
```

Comment:
Positive and NonNegative respectively to reflect the new argument checks. Pull the change to NonVanillaCustomItemData as well

Reply:
Added with the latest commit. I also improved the exception messages in the implementation of these methods.